File: gojek/tech-intern/maze.py
t = int(input())
case = 1

while t != 0:
    rows = int(input())
    cols = int(input())

    matrix = []
    for row in range(rows):
        matrix.append(list(input()))

    contested = 0
    storeChar = {}
    seenChar = []
    visited = set()

    def dfs(r, c, char):
        if (r < 0 or c < 0 or 
            r == rows or c == cols or 
            matrix[r][c] == "#" or 
            (r, c) in visited):
            return

        visited.add((r, c))

        if matrix[r][c] != "." and matrix[r][c] != char:
            seenChar.append(matrix[r][c])

        dfs(r + 1, c, char)
        dfs(r - 1, c, char)
        dfs(r, c + 1, char)
        dfs(r, c - 1, char)

    for r in range(rows):
        for c in range(cols):
            if matrix[r][c] != "#" and matrix[r][c] != "." and (r, c) not in visited:
                dfs(r, c, matrix[r][c])

                if len(seenChar) == 0:
                    if matrix[r][c] not in storeChar:
                        storeChar[matrix[r][c]] = 1
                    else:
                        storeChar[matrix[r][c]] += 1
                    
                if len(seenChar) > 0:
                    contested += 1

                seenChar = []

    # sort alphabetically
    alphabeticStoreChar = {k: v for k, v in sorted(storeChar.items())}

    print(f"Case {case}")
    for key in alphabeticStoreChar:
        print(key, alphabeticStoreChar[key])
    print("contested", contested)

    case += 1
    t -= 1

File: gojek/tech-intern/maze.cpp
#include <iostream>
#include <map>
#include <vector>

using namespace std;

int ROWSS;
int COLSS;

void dfs(int matrix[ROWSS][COLSS], int r, int c, char currChar, vector<int> seenChar, int rows, int cols)
{
    if (r < 0 || c < 0 || r == rows || c == cols || matrix[r][c] == '#' || matrix[r][c] == true)
    {
        return
    }

    matrix[r][c] = true;

    if (matrix[r][c] != "." && matrix[r][c] != currChar)
    {
        seenChar.push_back(matrix[r][c])
    }

    dfs(matrix, r + 1, c, currChar, seenChar, rows, cols);
    dfs(matrix, r - 1, c, currChar, seenChar, rows, cols);
    dfs(matrix, r, c + 1, currChar, seenChar, rows, cols);
    dfs(matrix, r, c - 1, currChar, seenChar, rows, cols);

    return;
}

int main()
{

    int t;
    cin >> t;

    while (t--)
    {
        int rows, cols;
        cin >> rows;
        cin >> cols;

        ROWSS = rows;
        COLSS = cols;

        string matrix[rows][cols];

        for (int r = 0; r < rows; r++)
        {
            string inputRow;
            cin >> inputRow;
            for (int c = 0; c < inputRow.length(); c++)
            {
                matrix[r][c] = inputRow[r];
            }
        }

        int contested = 0;
        map<string, int> storeChar;
        vector<int> seenChar, currChar;

        for (int r = 0; r < rows; r++)
        {
            for (int c = 0; c < cols; c++)
            {
                if(matrix[r][c] != true and matrix[r][c] != "#" and matrix[r][c] != "."){
                    dfs(matrix, r, c, currChar, seenChar, rows, cols);

                    cout << seenChar.size() << endl;
                }
            }
            cout << endl;
        }
    }

    return 0;
}
